### PR TITLE
[NTOS:EX][NTOS:MM] Minor code improvements, trivial change of behavior only, based on CORE-11958 patch

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Commit good parts of the abandoned patch,
plus some related improvements.

JIRA issue: [CORE-11958](https://jira.reactos.org/browse/CORE-11958)

Co-authored-by: "vshcherbyna"

## Proposed changes

Minor code improvements, trivial change of behavior only.

- Move a few RtlZeroMemory() calls around.
- Match a few variable name/types.
- A few trivial code rewrites.

## TODO

- [ ] Are the 2 'TODO?' code wanted, to match other places?

### CODE REVIEW

File: ntoskrnl/ex/handle.c [hunk 1]
```diff
@@ -728,16 +728,11 @@ ExpAllocateHandleTableEntry(IN PHANDLE_TABLE HandleTable,
...
-                {
-                    /* We're still the only thread around, so fail */
-                    NewHandle->GenericHandleOverlay = NULL;
-                    return NULL;
-                }
+            /* Check if allocation failed and nobody else went through here */
+            if (!Result && !OldValue)
+                /* We're still the only thread around, so fail */
+                NewHandle->GenericHandleOverlay = NULL;
+                return NULL;
```
Comment (learn-more) [inline ntoskrnl/ex/handle.c:735]: You probably introduced a bug here.

**Review by learn-more — CHANGES_REQUESTED** [review]
'Minor code improvements, trivial change of behavior' does not at all describe the purpose of a PR.

This is the last time I am going to ask you:
- If you create a PR, please describe the PURPOSE of the PR, not the literal code or some generic statements.

**Comment by JoachimHenze** [comment]
I do think the part of "moving downwards the 2 calls to RtlZeroMemory(Buffer, Size);" of this PR is fine and worth committing.

Any other change done in here, I do not understand and would request a convincing justification for. If you cannot give such convincing justification, they should be reverted.
That includes your added comments about where you raised the question "whether more calls of RtlZeroMemory would be beneficial". Either you are sure about that, and you can prove them to be missing, then do so and prove it.
Or you have no clue about ít, in that case it makes no sense to lure other people into the same thinking by adding those comments.

Same for the added casts: Does any compiler complain, or is proven to do the implicit cast wrong? If not, I see no benefit in adding them.

File: ntoskrnl/ex/handle.c (hunk 1 above)
Comment (katahiromz) [reply ntoskrnl/ex/handle.c:735]: { }

**Comment by binarymaster** [comment]
Also the build is rekt.

File: ntoskrnl/ex/handle.c (hunk 1 above)
Comment (SergeGautherie) [reply ntoskrnl/ex/handle.c]: Sure. I re-added the `{` line which was unintentionaly removed.

**Comment by SergeGautherie** [comment]
> added casts

gcc-i386-Debug, without casts:
```
2022-11-24T02:01:06.4836853Z ../src/ntoskrnl/mm/ARM3/virtual.c: In function 'NtAllocateVirtualMemory':
2022-11-24T02:01:06.4837490Z ../src/ntoskrnl/mm/ARM3/virtual.c:4797:32: error: passing argument 1 of 'MiInsertVadEx' from incompatible pointer type [-Werror=incompatible-pointer-types]
2022-11-24T02:01:06.4837911Z          Status = MiInsertVadEx(Vad,
2022-11-24T02:01:06.4838135Z                                 ^~~
2022-11-24T02:01:06.4838390Z In file included from ../src/ntoskrnl/mm/ARM3/virtual.c:16:
2022-11-24T02:01:06.4838939Z ../src/ntoskrnl/mm/ARM3/miarm.h:2237:20: note: expected 'PMMVAD' {aka 'struct _MMVAD *'} but argument is of type 'PMMVAD_LONG' {aka 'struct _MMVAD_LONG *'}
2022-11-24T02:01:06.4839286Z      _Inout_ PMMVAD Vad,
2022-11-24T02:01:06.4839483Z              ~~~~~~~^~~
2022-11-24T02:01:06.4839821Z ../src/ntoskrnl/mm/ARM3/virtual.c: In function 'NtFreeVirtualMemory':
2022-11-24T02:01:06.4840408Z ../src/ntoskrnl/mm/ARM3/virtual.c:5595:33: error: passing argument 1 of 'MiInsertVad' from incompatible pointer type [-Werror=incompatible-pointer-types]
2022-11-24T02:01:06.4840894Z                      MiInsertVad(NewVad, &Process->VadRoot);
2022-11-24T02:01:06.4841136Z                                  ^~~~~~
2022-11-24T02:01:06.4841381Z In file included from ../src/ntoskrnl/mm/ARM3/virtual.c:16:
2022-11-24T02:01:06.4841907Z ../src/ntoskrnl/mm/ARM3/miarm.h:2231:20: note: expected 'PMMVAD' {aka 'struct _MMVAD *'} but argument is of type 'PMMVAD_LONG' {aka 'struct _MMVAD_LONG *'}
2022-11-24T02:01:06.4842240Z      _Inout_ PMMVAD Vad,
2022-11-24T02:01:06.4842436Z              ~~~~~~~^~~
```

**Comment by ThFabba** [comment]
I am against moving these `RtlZeroMemory` calls. They belong right after (or inside, with ExAllocatePoolZero) the allocation for clarity and to avoid future bugs. That is significantly more important than optimizing the performance of error cases.

**Comment by JoachimHenze** [comment]
Ok Thomas, I do trust your educated judgement more than my own feeling regarding the moving of the RtlZeroMemory calls.
I am fine with your opinion superseding mine here therefore. Follow Thomas advice then.

Are you fine with the casts to silence the GCC build? Serges justification looks legit to me, but I cannot judge the correctness of those changes,

**Comment by ThFabba** [comment]
The casts are necessary if we're doing the type change. Why was the type change necessary? 🤷

**Pull request closed without merging**